src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/UseStringEqualsOverStringCompare.cs

```diff
@@ -0,0 +1,248 @@
+
+        /// <summary>
+        /// Returns true if the specified <see cref="IBinaryOperation"/>:
+        /// <list type="bullet">
+        /// <item>Is an equals or not-equals operation</item>
+        /// <item>One operand is a literal zero</item>
+        /// <item>The other operand is an invocation of <see cref="string.Compare(string, string, bool)"/></item>
+        /// <item>The <c>ignoreCase</c> argument is a boolean literal</item>
+        /// </list>
+        /// </summary>
+        /// <param name="binaryOperation"></param>
+        /// <param name="symbols"></param>
+        /// <returns></returns>
+        internal static bool IsStringStringBoolCase(IBinaryOperation binaryOperation, RequiredSymbols symbols)
+        {
+            var invocation = GetInvocationFromEqualityCheckWithLiteralZero(binaryOperation);
+
+            return invocation is not null &&
+                //  Don't report a diagnostic if either the string.Compare overload or the
+                //  corrasponding string.Equals overload is missing.
+                symbols.CompareStringStringBool is not null &&
+                symbols.EqualsStringStringStringComparison is not null &&
+                invocation.TargetMethod.Equals(symbols.CompareStringStringBool) &&
+                //  Only report a diagnostic if the 'ignoreCase' argument is a boolean literal.
+                invocation.Arguments.GetArgumentForParameterAtIndex(2).Value is ILiteralOperation literal &&
+                literal.ConstantValue.Value is bool;
+        }
+
+        /// <summary>
+        /// Returns true if the specified <see cref="IBinaryOperation"/>:
+        /// <list type="bullet">
+        /// <item>Is an equals or not-equals operation</item>
+        /// <item>One operand is a literal zero</item>
+        /// <item>The other operand is any invocation of <see cref="string.Compare(string, string, StringComparison)"/></item>
+        /// </list>
+        /// </summary>
+        /// <param name="binaryOperation"></param>
+        /// <param name="symbols"></param>
+        /// <returns></returns>
+        internal static bool IsStringStringStringComparisonCase(IBinaryOperation binaryOperation, RequiredSymbols symbols)
+        {
+            var invocation = GetInvocationFromEqualityCheckWithLiteralZero(binaryOperation);
+
+            return invocation is not null &&
+                //  Don't report a diagnostic if either the string.Compare overload or the
+                //  corrasponding string.Equals overload is missing.
+                symbols.CompareStringStringStringComparison is not null &&
+                symbols.EqualsStringStringStringComparison is not null &&
+                invocation.TargetMethod.Equals(symbols.CompareStringStringStringComparison, SymbolEqualityComparer.Default);
+        }
+
+        //  No IOperation instances are being stored here.
+#pragma warning disable RS1008 // Avoid storing per-compilation data into the fields of a diagnostic analyzer
```

Comment:
Is there a bug for this false positive on RS1008?

Reply:
[Bug report](https://github.com/dotnet/roslyn-analyzers/issues/782)